### ISSUE TITLE
MIN/MAX(VARCHAR)

### ISSUE THREAD

**Issue by hawkfish** [cwida/duckdb#265]
It looks like these aggregates return NULL for string arguments. Is that because the comparisons might need collation so this has been postponed?

**Comment by Mytherin** [comment]
Good catch, it looks like this is a bug. Looks like both the grouped and the non-grouped MIN/MAX aggregates don't work properly on strings. Looks like this was an oversight there.

**Comment by hawkfish** [comment]
I noticed this because in the work on `STRING_AGG` I was trying to figure out how to manage non-scalar state in an aggregate. I went to look at these two and realised they weren't implemented, so I ended up using the string heap of the first input. That seemed to work for `STRING_AGG` but I was heading towards `ARRAY_AGG` as a basis for implementing holistic aggregates like `MEDIAN`, `MAD` and  `GINI` (as in coefficient).

My use of the string heap seemed to work, but I'd appreciate it if someone else more familiar with the data model could think about these two and make sure this is the right approach.
